storage/remote/codec.go

```diff
@@ -160,6 +160,123 @@ func FromQueryResult(sortSeries bool, res *prompb.QueryResult) storage.SeriesSet
 	}
 }
 
+// chunkedResponseSeriesSet implements storage.SeriesSet.
+type chunkedResponseSeriesSet struct {
+	stream        *ChunkedReader
+	currentChunks []prompb.Chunk
+	currentLabels []prompb.Label
+	httpResp      *http.Response
+}
+
+func (c *chunkedResponseSeriesSet) Next() bool {
+	c.currentLabels = nil
+	c.currentChunks = nil
+	res := &prompb.ChunkedReadResponse{}
+	if err := c.stream.NextProto(res); err != nil {
+		if c.httpResp != nil {
+			c.httpResp.Body.Close()
+		}
+		return false
+	}
+	c.currentLabels = res.ChunkedSeries[0].Labels
+	c.currentChunks = append(c.currentChunks, res.ChunkedSeries[0].Chunks...)
+	return true
+}
+
+func (c *chunkedResponseSeriesSet) At() storage.Series {
+	return &chunkSeries{
+		labels: prompb.Labels{Labels: c.currentLabels},
+		chunks: c.currentChunks,
+	}
+}
+
+func (c *chunkedResponseSeriesSet) Warnings() storage.Warnings { return nil }
+
+func (c *chunkedResponseSeriesSet) Err() error {
+	return nil
+}
+
+// chunkSeries implements storage.Series
+type chunkSeries struct {
+	labels     prompb.Labels
+	chunks     []prompb.Chunk
+	mint, maxt int64
+}
+
+func (c *chunkSeries) Labels() labels.Labels {
+	return labelProtosToLabels(c.labels.Labels)
+}
+
+func (c *chunkSeries) Iterator() chunkenc.Iterator {
+	streamingChunkSeriesIterator := &chunkSeriesIterator{chunks: c.chunks, i: 0, mint: c.mint, maxt: c.maxt}
+	_ = streamingChunkSeriesIterator.resetCurIterator()
+	return streamingChunkSeriesIterator
+}
+
+type chunkSeriesIterator struct {
+	// Index poiting to a specific chunk in []prompb.Chunk.
+	i int
+	// Current chunks iterator.
+	cur        chunkenc.Iterator
+	maxt, mint int64
+	chunks     []prompb.Chunk
+}
+
+func (it *chunkSeriesIterator) resetCurIterator() error {
+	currectChunk := it.chunks[it.i]
+	it.mint = currectChunk.MinTimeMs
+	it.maxt = currectChunk.MaxTimeMs
+	chk, err := chunkenc.FromData(chunkenc.Encoding(currectChunk.Type), currectChunk.Data)
+	if err != nil {
+		return err
+	}
+	it.cur = chk.Iterator(nil)
+	return nil
```

Comment:
Why we return error if we never use it?

Reply:
Oops, I misunderstood this comment and removed the return statement altogether which caused some more confusion on 'ignoring comments/review' below. Sorry about that.
It's been fixed!